doc/chapters/gateways/10-environment-gateway.rst

```diff
@@ -74,4 +74,14 @@ to e.g. ``/tmp/test``, the varaiable value would now be set to ``/tmp/test:/some
 If ``SOME_ENVIRONMENT_VARIABLE`` had not been previously set, the value would now be
 set to ``:/some/path``.
 
+Important Note
+--------------
 
+Environment variables are internally stored as an array of pointers to strings in Linux based
+systems. This array is terminated by a NULL pointer. Each name and value pairs are matched with "="
+sign and stored together. For instance the first example configuration above will be stored as
+``SOME_ENVIRONMENT_VARIABLE=SOME_VALUE`` in environment table. Thus, please note that it would be
+necessary to **refrain from setting environment variable names or values anything that includes**
+
+- the equal sign ``'='``
+- embedded NULL characters ``'\0'``.
```

Comment:
Of course should i also add thing about other variables which are used setuid or library path or debug path.  For instance PATH, LD_LIBRARY_PATH, LD_DEBUG, LD_PRELOAD.